lecture_4/homework/app/models/null_weapon.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class NullWeapon
+  def type
+    'bare hands'
+  end
+
+  def self.create!(*args); end
+end
```

Comment:
Zastosowanie NullObjectu dla broni. Dzięki temu metoda attack wojowników nie podnosi wyjątku. Plus jeżeli z jakiegoś powodu preferred_weapon_kind najemnika nie będzie z żadnego z trzech obsłużonych typów (melee, ranged i magical) to zamiast podnieść wyjątek WeaponCreator zwróci nil.